src/Symfony/Component/Validator/Tests/Constraints/TimeValidatorTest.php

```diff
@@ -62,6 +69,50 @@ public function getValidTimes()
         ];
     }
 
+    /**
+     * @dataProvider getValidTimesWithoutSeconds
+     */
+    public function testValidTimesWithoutSeconds($time)
+    {
+        $this->validator->validate($time, new Time([
+            'withSeconds' => false,
+        ]));
+
+        $this->assertNoViolation();
+    }
+
+    public function getValidTimesWithoutSeconds()
+    {
+        return [
+            ['01:02'],
+            ['00:00'],
+            ['23:59'],
+        ];
+    }
+
+
+    /**
+     * @dataProvider getInvalidTimesWithoutSeconds
+     */
+    public function testInvalidTimesWithoutSeconds($time)
+    {
+        $this->validator->validate($time, $constraint = new Time());
+
+        $this->buildViolation($constraint->message)
+            ->setParameter('{{ value }}', '"'.$time.'"')
+            ->setCode(Time::INVALID_FORMAT_ERROR)
+            ->assertRaised();
+    }
+
+    public function getInvalidTimesWithoutSeconds()
```

Comment:
Same as above, I would use `yield` and the `\Generator` return type for this one too